addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,370 @@
+
+    // -------------------------------------------------------------------------
+    // Private
+    // -------------------------------------------------------------------------
+
+    // EVENTS
+    // -------------------------------------------------------------------------
+    _onChangeRadio: function (event) {
+        var $radioGroup = $(event.currentTarget).parents('.o_survey_form_simple_radio');
+        return this._updateSimpleRadioForDisplay($radioGroup);
+    },
+
+    _onChangeSelect: function (event) {
+        var $selectGroup = $(event.currentTarget).parents('.o_survey_form_simple_select');
+        return this._updateSimpleSelectForDisplay($selectGroup);
+    },
+
+    _onChangeMultipleCheckbox: function (event) {
+        var $checkGroup = $(event.currentTarget).parents('.o_survey_form_multiple');
+        return this._updateMultipleChoiceForDisplay($checkGroup);
+    },
+
+    _onBreadcrumbClick: function (event) {
+        var self = this;
+        event.preventDefault();
+        $('<input>').attr({type: 'hidden', name: 'breadcrumb_redirect', value: $valObj.attr('href')}).appendTo('.js_surveyform');
+
+        self.$el.find('button[name="button_submit"]').click();
+//        $('.js_surveyform').submit();
+    },
+
+    _onSubmit: function (event) {
+        var self = this;
+        event.preventDefault();
+        var $target = $(event.currentTarget);
+        var submitValue = $target.val();
+        var $form = $target.parents('form.o_survey_form');
+        var formData = new FormData($form[0]);
+        var params = {
+            button_submit: submitValue,
+        };
+
+        this.$('div.o_survey_form_date').each(function () {
+            self._updateDateForSubmit($(this), formData);
+        });
+        this._prepareSubmitValues(formData, params);
+
+        this._resetErrors();
+
+        return self._rpc({
+            route: '/survey/submit/' + self.surveyToken + '/' + self.answerToken,
+            params: params,
+        }).then(function (result) {
+            return self._onSubmitDone(result, params);
+        });
+    },
+
+    _onSubmitDone: function (result, params) {
+        var self = this;
+        if (result && !result.error) {
+            window.location = result;
+        }
+        else if (result && result.fields && result.error === 'validation') {
+            var fieldKeys = _.keys(result.fields);
+            _.each(fieldKeys, function (key) {
+                self.$("#" + key + '>.o_survey_question_error').append($('<p>', {text: result.fields[key]})).show();
+                if (fieldKeys[fieldKeys.length - 1] === key) {
+                    self._scrollToError(self.$('.o_survey_question_error:visible:first').closest('.js_question-wrapper'));
+                }
+            });
+            return false;
+        }
+        else {
+            var $target = self.$('.o_survey_error');
+            $target.append('<p>Their was an error during the validation of the survey.</p>').show();
```

Comment:
"There was" + needs to be translated.

Suggestion: `$('<p/>', {text: _('There was an error during the validation of the survey.')})`